core-java-modules/core-java-string-algorithms-3/src/test/java/com/baeldung/repeatedcharstring/RepeatedCharacterStringUnitTest.java

```diff
@@ -0,0 +1,136 @@
+package com.baeldung.repeatedcharstring;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.nio.CharBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+
+import static ch.qos.logback.core.CoreConstants.EMPTY_STRING;
+import static java.util.stream.Stream.generate;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class RepeatedCharacterStringUnitTest {
+
+    @Test
+    public void givenString_whenStringBufferUsed_thenStringCreated() {
+        String exampleString = "aaaaaaa";
+        int length = 7;
+        char charToAppend = 'a';
+        StringBuilder outputBuffer = new StringBuilder(length);
```

Comment:
`outputBuffer` is _slightly_ confusing a name here. There is a `StringBuffer` class, which we're NOT using. `outputBuilder` might be clearer.

Reply:
resolved